lib/logstash/filters/dictionary/csv_file.rb

```diff
@@ -0,0 +1,22 @@
+# encoding: utf-8
+require "csv"
+
+module LogStash module Filters module Dictionary
+  class CsvFile < File
+    def initialize_for_file_type
+      @io = StringIO.new("")
+      @csv = CSV.new(@io)
+    end
+
+    def read_file_into_dictionary
+      # low level CSV read that tries to create as
+      # few intermediate objects as possible
+      # this overwrites the value at key
+      IO.foreach(@dictionary_path, :mode => 'r:bom|utf-8') do |line|
+        @io.string = line
```

Comment:
not sur eI understand the purpose of this `@io` here?

Reply:
ah yes, I missed the  `CSV.new(@io)`. 

So I'd be curious to actually measure the actual GC impact here. I personally don't think it would make any noticeable difference. IMO such GC related optimizations are usually not necessary unless we are dealing with very hot code paths. For code outside the hot paths I'd suggest code correctness and readability. 
OTOH sometimes simple changes can actually avoid superfluous object creation which can be beneficial if there is a valid reason for such specific optimizations.

Reply:
Bear in mind that I am optimising for very large files.